### PR TITLE
support turbomind head_dim 64

### PR DESCRIPTION
**Pull request by irexyc**
## Motivation

support models with head_dim = 64 like `InternVL/InternVL2-1B/` and `Qwen/Qwen1.5-0.5B-Chat/`

### CODE REVIEW

File: lmdeploy/turbomind/deploy/source_model/internvl.py [hunk 1]
```diff
@@ -80,8 +80,14 @@ def model_info(self):
                 scaling_factor = model_arg['rope_scaling'].get('factor', '')
                 if scaling_type == 'dynamic':
                     use_dynamic_ntk = 1
+            addition_kwargs = {}
```
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/internvl.py:83]: `addition_kwargs` may not a good option. Please use `attn_bias` intentionally.
```python
attn_bias = 1 if model_arg['architectures'][0] == 'Qwen2ForCausalLM' else 0
```

File: lmdeploy/turbomind/supported_models.py [hunk 2]
```diff
@@ -65,11 +65,11 @@ def is_supported(model_path: str):
     """  # noqa: E501
     import os
 
-    def _is_head_dim_128(cfg):
+    def _is_head_dim_128_64(cfg):
```
Comment (lvhan028) [inline lmdeploy/turbomind/supported_models.py:68]: Let's use `_is_head_dim_supported(cfg)` instead

File: src/turbomind/kernels/attention/decoding.cu [hunk 3]
```diff
@@ -73,31 +72,41 @@ void dispatchDecoding(const AttentionParams<T>& params)
...
+            return dispatch_h(arch, uint8_t{}, dim);
         }
         else {
-            return dispatch_h(arch, T{});
+            return dispatch_h(arch, T{}, dim);
+        }
+        return false;
+    };
+
+    auto dispatch_d = [&](auto arch) {
```
Comment (lvhan028) [inline src/turbomind/kernels/attention/decoding.cu:89]: what `d` means in `dispatch_d`?

File: src/turbomind/kernels/attention/kv_cache_utils_v2.cu [hunk 4]
```diff
@@ -241,10 +241,10 @@ void invokeProcessKV_v2(char**       blocks,
     int  block = WARPS * WARP_SIZE;
     dim3 grid((max_q_len + CTA_S - 1) / CTA_S, head_num, batch_size);
 
-    auto invoke = [&](auto tkv) {
+    auto invoke = [&](auto tkv, const auto dim) {
```
Comment (lvhan028) [inline src/turbomind/kernels/attention/kv_cache_utils_v2.cu:244]: @lzhangzz what does `tkv` mean?

Comment (irexyc) [reply src/turbomind/kernels/attention/kv_cache_utils_v2.cu:244]: I think it means target kv datatype

File: lmdeploy/turbomind/deploy/source_model/internvl.py [hunk 5]
```diff
@@ -80,8 +80,13 @@ def model_info(self):
                 scaling_factor = model_arg['rope_scaling'].get('factor', '')
                 if scaling_type == 'dynamic':
                     use_dynamic_ntk = 1
+            attn_bias = 1 if model_arg['architectures'][
+                0] == 'Qwen2ForCausalLM' else 0
 
         return dict(num_layer=num_layer,
+                    size_per_head=hidden_units // attn_head_num,
```
Comment (lvhan028) [inline lmdeploy/turbomind/deploy/source_model/internvl.py:87]: May also check whether other source models miss these two parameters

Comment (irexyc) [reply lmdeploy/turbomind/deploy/source_model/internvl.py:87]: other models (glm4, deepseek_vl) don't have config with head_dim=64

**Comment by lvhan028** [comment]
@zhulinJulia24 may add the following models into tm test set
- meta-llama/Llama-3.2-1B-Instruct
- Qwen/Qwen2.5-0.5B-Instruct
- InternVL/InternVL2-1B

**Review by lvhan028 — APPROVED** [review]

**Comment by zhulinJulia24** [comment]
done

> @zhulinJulia24 may add the following models into tm test set
> 
> * meta-llama/Llama-3.2-1B-Instruct
> * Qwen/Qwen2.5-0.5B-Instruct
> * InternVL/InternVL2-1B

done

**Review by lzhangzz — APPROVED** [review]

**Pull request merged**